File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/java/src/test/java/example/micronaut/BookInventoryClientStub.java
package example.micronaut;

import io.micronaut.context.annotation.Requires;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.retry.annotation.Fallback;
import jakarta.inject.Singleton;
import reactor.core.publisher.Mono;

import jakarta.validation.constraints.NotBlank;

import static io.micronaut.context.env.Environment.TEST;

@Requires(env = TEST) // <1>
@Fallback
@Singleton
public class BookInventoryClientStub implements BookInventoryOperations {

    @Override
    public Mono<Boolean> stock(@NonNull @NotBlank String isbn) {
        if (isbn.equals("1491950358")) {
            return Mono.just(true); // <2>
        }
        if (isbn.equals("1680502395")) {
            return Mono.just(false); // <3>
        }
        return Mono.empty(); // <4>
    }
}

File: guides/micronaut-microservices-services-discover-consul/bookrecommendation/kotlin/src/test/kotlin/example/micronaut/BookInventoryClientStub.kt
package example.micronaut

import io.micronaut.context.annotation.Requires
import io.micronaut.context.env.Environment.TEST
import io.micronaut.retry.annotation.Fallback
import jakarta.inject.Singleton
import reactor.core.publisher.Mono
import jakarta.validation.constraints.NotBlank

@Requires(env = [TEST]) // <1>
@Fallback
@Singleton
open class BookInventoryClientStub : BookInventoryOperations {

    override fun stock(@NotBlank isbn: String): Mono<Boolean> {
        if (isbn == "1491950358") {
            return Mono.just(true) // <2>
        }
        if (isbn == "1680502395") {
            return Mono.just(false) // <3>
        }
        return Mono.empty() // <4>
    }
}
